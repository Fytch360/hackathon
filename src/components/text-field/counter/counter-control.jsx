import React from 'react'
import PropTypes from 'prop-types'
import { extend, omit } from 'lodash'

import { disableHandler, preventHandler } from '../../utils/handlers'
import { plus, minus } from '../../icon/common'

import { CounterButtonStyled, IconStyled } from './text-field-counter.style'

const iconMap = {
    increase: plus,
    decrease: minus
}

export const CounterControl = ({ onClick, disabled, mode, ...rest }) => (
    <CounterButtonStyled
        type="button"
        onClick={disableHandler(preventHandler(onClick), disabled)}
        disabled={disabled}
        tabIndex={-1}
        {...rest}
    >
        <IconStyled icon={iconMap[mode]} />
    </CounterButtonStyled>
)

CounterControl.propTypes = {
    mode: PropTypes.oneOf(['increase', 'decrease']).isRequired,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
}
