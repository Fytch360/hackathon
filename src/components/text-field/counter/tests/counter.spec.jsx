import React from 'react'
import { mount } from 'enzyme'

import { TextFieldCounter } from '..'
import { CounterControl } from '../counter-control'

const ARROW_UP_KEY_CODE = 38
const ARROW_DOWN_KEY_CODE = 40
const HOME_BUTTON_KEY_CODE = 36
const END_BUTTON_KEY_CODE = 35
const PAGE_UP_BUTTON_KEY_CODE = 33
const PAGE_DOWN_BUTTON_KEY_CODE = 34

describe('<TextFieldCounter />', () => {

    test('Is available', () => {
        expect(TextFieldCounter).toBeDefined()
    })

    test('Snapshot', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 0,
            max: 8,
            step: 1,
            value: '2',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        expect(wrapper).toMatchSnapshot()
    })

    test('Should render 2 CounterControls', () => {
        const wrapper = mount(<TextFieldCounter />)

        expect(wrapper.find(CounterControl).length).toBe(2)
    })

    test('Should call onChange func if counter controls was clicked', () => {
        const onChangeMock = jest.fn()

        const propsExample = {
            min: 0,
            max: 8,
            step: 1,
            value: '2',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        expect(onChangeMock.mock.calls.length).toBe(0)

        wrapper.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls.length).toBe(1)

        wrapper.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls.length).toBe(2)
    })

    test('Should decrease value if decrease button was clicked', () => {
        const onChangeMock = jest.fn()

        const propsExample = {
            min: 0,
            max: 8,
            step: 1,
            value: '2',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('1')
    })

    test('Should increase value if increase button was clicked', () => {
        const onChangeMock = jest.fn()

        const propsExample = {
            min: 0,
            max: 8,
            step: 1,
            value: '2',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('3')
    })

    test('Should increase value on any step', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 0,
            max: 36,
            value: '2',
            onChange: onChangeMock
        }

        const stepValue1 = 6
        const wrapper1 = mount(<TextFieldCounter {...propsExample} step={stepValue1} />)

        wrapper1.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('8')

        const stepValue2 = 10
        const wrapper2 = mount(<TextFieldCounter {...propsExample} step={stepValue2} />)

        wrapper2.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[1][0]).toBe('12')
    })

    test('Should decrease value on any step', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 0,
            max: 36,
            value: '16',
            onChange: onChangeMock
        }

        const stepValue1 = 6
        const wrapper1 = mount(<TextFieldCounter {...propsExample} step={stepValue1} />)

        wrapper1.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('10')

        const stepValue2 = 10
        const wrapper2 = mount(<TextFieldCounter {...propsExample} step={stepValue2} />)

        wrapper2.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[1][0]).toBe('6')
    })

    test('Should set max value if increase button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 4,
            value: '18',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('20')
    })


    test('Should set min value if decrease button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 8,
            value: '7',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls[0][0]).toBe('5')
    })

    test('Should\'nt call on Change if counter is disabled', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 16,
            step: 2,
            value: '12',
            onChange: onChangeMock,
            disabled: true
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('button').first()
            .simulate('click', {})

        wrapper.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls.length).toBe(0)
    })

    test('Should\'nt call onChange if value equals limit value', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 16,
            step: 4,
            onChange: onChangeMock
        }

        const wrapper1 = mount(<TextFieldCounter {...propsExample} value="16" />)

        wrapper1.find('button').last()
            .simulate('click', {})

        expect(onChangeMock.mock.calls.length).toBe(0)

        const wrapper2 = mount(<TextFieldCounter {...propsExample} value="5" />)

        wrapper2.find('button').first()
            .simulate('click', {})

        expect(onChangeMock.mock.calls.length).toBe(0)
    })

    test('Should correctly change value if arrow up was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: ARROW_UP_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('16')
    })

    test('Should correctly change value if arrow down was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: ARROW_DOWN_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('12')
    })

    test('Should correctly change value if home button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: HOME_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('5')
    })

    test('Should set zero value if home button was pressed without min value', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            max: 20,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: HOME_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('0')
    })

    test('Should correctly change value if end button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 20,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: END_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('20')
    })

    test('Should\'nt call onChange if end button was pressed without max value', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: END_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls.length).toBe(0)
    })

    test('Should correctly change value if page up button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 100,
            step: 2,
            value: '14',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: PAGE_UP_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('34')
    })

    test('Should correctly change value if page up button was pressed', () => {
        const onChangeMock = jest.fn()
        const propsExample = {
            min: 5,
            max: 100,
            step: 2,
            value: '53',
            onChange: onChangeMock
        }

        const wrapper = mount(<TextFieldCounter {...propsExample} />)

        wrapper.find('input')
            .simulate('keydown', { keyCode: PAGE_DOWN_BUTTON_KEY_CODE })

        expect(onChangeMock.mock.calls[0][0]).toBe('33')
    })
})
