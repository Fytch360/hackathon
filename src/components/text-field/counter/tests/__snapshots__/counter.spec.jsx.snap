// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldCounter /> Snapshot 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border: solid 1px;
  border-radius: 4px;
  -webkit-transition: 0.17s ease box-shadow;
  transition: 0.17s ease box-shadow;
  height: 54px;
  padding: 0;
  padding-right: 16px;
  padding-left: 16px;
}

.emotion-0:hover {
  border: solid 1px;
}

.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  border-radius: 4px;
  width: 100%;
  outline: none;
  display: block;
  font-weight: 500;
  background-color: transparent;
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  -ms-appearance: textfield;
  appearance: textfield;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  margin-right: 16px;
  font-size: 15px;
  line-height: 20px;
}

.emotion-2::-ms-clear {
  display: none;
}

.emotion-2::-webkit-inner-spin-button,
.emotion-2::-webkit-outer-spin-button,
.emotion-2::-webkit-clear-button,
.emotion-2::-webkit-inner-spin-button,
.emotion-2::-webkit-calendar-picker-indicator {
  display: none;
}

.emotion-2:-webkit-autofill,
.emotion-2:-webkit-autofill:hover,
.emotion-2:-webkit-autofill:focus,
.emotion-2:-webkit-autofill:active {
  -webkit-text-fill-color: inherit;
  box-shadow: 0 0 0 1000px transparent inset;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.emotion-4 {
  padding-left: 12px;
  border-left: 1px solid;
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: inline-block;
  padding: 0;
  padding-right: 16px;
  padding-left: 16px;
}

.emotion-4 .emotion-7 {
  margin: 0;
  margin-right: 8px;
}

.emotion-4 .emotion-7:last-child {
  margin-right: 0;
}

.emotion-6 {
  outline: none;
  cursor: pointer;
  border-radius: 4px;
  border: none;
  width: 36px;
  height: 36px;
  padding: 0;
  -webkit-transition: background-color 0.17s ease-in-out,box-shadow 0.17s ease-in-out,border-color 0.17s ease-in-out;
  transition: background-color 0.17s ease-in-out,box-shadow 0.17s ease-in-out,border-color 0.17s ease-in-out;
}

.emotion-6 .emotion-9 svg {
  width: 16px;
  height: 16px;
}

.emotion-6:disabled {
  cursor: default;
}

.emotion-13 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
}

.emotion-13 svg {
  display: block;
}

<OmittedComponent
  max={8}
  min={0}
  onChange={[MockFunction]}
  step={1}
  value="2"
>
  <Component
    max={8}
    min={0}
    onChange={[MockFunction]}
    step={1}
    value="2"
  >
    <TextFieldMasked
      additionalChild={
        <CounterStyled
          size="md"
        >
          <CounterControl
            aria-hidden="true"
            disabled={false}
            mode="decrease"
            onClick={[Function]}
          />
          <CounterControl
            aria-hidden="true"
            disabled={false}
            mode="increase"
            onClick={[Function]}
          />
        </CounterStyled>
      }
      aria-valuemax={8}
      aria-valuemin={0}
      aria-valuenow="2"
      aria-valuetext="undefined2undefined"
      autoComplete="off"
      inputMode="decimal"
      maskOptions={
        Object {
          "blocks": Object {
            "num": Object {
              "mask": [Function],
              "max": 8,
              "min": 0,
            },
          },
          "lazy": false,
          "mask": "num",
        }
      }
      onChange={[MockFunction]}
      onKeyDown={[Function]}
      role="spinbutton"
      size="md"
      value="2"
    >
      <IMask(Component)
        additionalChild={
          <CounterStyled
            size="md"
          >
            <CounterControl
              aria-hidden="true"
              disabled={false}
              mode="decrease"
              onClick={[Function]}
            />
            <CounterControl
              aria-hidden="true"
              disabled={false}
              mode="increase"
              onClick={[Function]}
            />
          </CounterStyled>
        }
        aria-valuemax={8}
        aria-valuemin={0}
        aria-valuenow="2"
        aria-valuetext="undefined2undefined"
        autoComplete="off"
        blocks={
          Object {
            "num": Object {
              "mask": [Function],
              "max": 8,
              "min": 0,
            },
          }
        }
        inputMode="decimal"
        lazy={false}
        mask="num"
        onAccept={[Function]}
        onBlur={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        role="spinbutton"
        size="md"
        unmask={true}
        value="2"
      >
        <Component
          additionalChild={
            <CounterStyled
              size="md"
            >
              <CounterControl
                aria-hidden="true"
                disabled={false}
                mode="decrease"
                onClick={[Function]}
              />
              <CounterControl
                aria-hidden="true"
                disabled={false}
                mode="increase"
                onClick={[Function]}
              />
            </CounterStyled>
          }
          aria-valuemax={8}
          aria-valuemin={0}
          aria-valuenow="2"
          aria-valuetext="undefined2undefined"
          autoComplete="off"
          defaultValue="2"
          inputMode="decimal"
          inputRef={[Function]}
          onBlur={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          role="spinbutton"
          size="md"
        >
          <OmittedComponent
            additionalChild={
              <CounterStyled
                size="md"
              >
                <CounterControl
                  aria-hidden="true"
                  disabled={false}
                  mode="decrease"
                  onClick={[Function]}
                />
                <CounterControl
                  aria-hidden="true"
                  disabled={false}
                  mode="increase"
                  onClick={[Function]}
                />
              </CounterStyled>
            }
            aria-valuemax={8}
            aria-valuemin={0}
            aria-valuenow="2"
            aria-valuetext="undefined2undefined"
            autoComplete="off"
            defaultValue="2"
            inputMode="decimal"
            onBlur={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            refWrapper={[Function]}
            role="spinbutton"
            size="md"
          >
            <Component
              additionalChild={
                <CounterStyled
                  size="md"
                >
                  <CounterControl
                    aria-hidden="true"
                    disabled={false}
                    mode="decrease"
                    onClick={[Function]}
                  />
                  <CounterControl
                    aria-hidden="true"
                    disabled={false}
                    mode="increase"
                    onClick={[Function]}
                  />
                </CounterStyled>
              }
              aria-valuemax={8}
              aria-valuemin={0}
              aria-valuenow="2"
              aria-valuetext="undefined2undefined"
              autoComplete="off"
              defaultValue="2"
              inputMode="decimal"
              onBlur={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              refWrapper={[Function]}
              role="spinbutton"
              size="md"
            >
              <WrapperStyled
                focused={false}
                horizontalPadding="inner"
                size="md"
                verticalPadding="inner"
              >
                <span
                  className="emotion-0 emotion-1"
                  size="md"
                >
                  <TextFieldStyled
                    aria-valuemax={8}
                    aria-valuemin={0}
                    aria-valuenow="2"
                    aria-valuetext="undefined2undefined"
                    autoComplete="off"
                    defaultValue="2"
                    horizontalMargin="inner"
                    horizontalMarginDirection="right"
                    horizontalPadding="inner"
                    inputMode="decimal"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    role="spinbutton"
                    size="md"
                    verticalPadding="inner"
                  >
                    <input
                      aria-valuemax={8}
                      aria-valuemin={0}
                      aria-valuenow="2"
                      aria-valuetext="undefined2undefined"
                      autoComplete="off"
                      className="emotion-2 emotion-3"
                      defaultValue="2"
                      inputMode="decimal"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      role="spinbutton"
                      size="md"
                    />
                  </TextFieldStyled>
                  <CounterStyled
                    size="md"
                  >
                    <span
                      className="emotion-4 emotion-5"
                      size="md"
                    >
                      <CounterControl
                        aria-hidden="true"
                        disabled={false}
                        mode="decrease"
                        onClick={[Function]}
                      >
                        <CounterButtonStyled
                          aria-hidden="true"
                          disabled={false}
                          onClick={[Function]}
                          tabIndex={-1}
                          type="button"
                        >
                          <button
                            aria-hidden="true"
                            className="emotion-6 emotion-7"
                            disabled={false}
                            onClick={[Function]}
                            tabIndex={-1}
                            type="button"
                          >
                            <IconStyled
                              icon="<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <rect x=\\"2\\" y=\\"7\\" width=\\"12\\" height=\\"2\\" rx=\\"1\\"></rect>    </g></svg>"
                            >
                              <Icon
                                className="emotion-8 emotion-9"
                                icon="<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <rect x=\\"2\\" y=\\"7\\" width=\\"12\\" height=\\"2\\" rx=\\"1\\"></rect>    </g></svg>"
                              >
                                <WrapperStyled
                                  className="emotion-8 emotion-9"
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <rect x=\\"2\\" y=\\"7\\" width=\\"12\\" height=\\"2\\" rx=\\"1\\"></rect>    </g></svg>",
                                    }
                                  }
                                  onClick={[Function]}
                                >
                                  <span
                                    className="emotion-9 emotion-13 emotion-14"
                                    dangerouslySetInnerHTML={
                                      Object {
                                        "__html": "<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <rect x=\\"2\\" y=\\"7\\" width=\\"12\\" height=\\"2\\" rx=\\"1\\"></rect>    </g></svg>",
                                      }
                                    }
                                    onClick={[Function]}
                                  />
                                </WrapperStyled>
                              </Icon>
                            </IconStyled>
                          </button>
                        </CounterButtonStyled>
                      </CounterControl>
                      <CounterControl
                        aria-hidden="true"
                        disabled={false}
                        mode="increase"
                        onClick={[Function]}
                      >
                        <CounterButtonStyled
                          aria-hidden="true"
                          disabled={false}
                          onClick={[Function]}
                          tabIndex={-1}
                          type="button"
                        >
                          <button
                            aria-hidden="true"
                            className="emotion-6 emotion-7"
                            disabled={false}
                            onClick={[Function]}
                            tabIndex={-1}
                            type="button"
                          >
                            <IconStyled
                              icon="<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <path d=\\"M8,2 C8.55228475,2 9,2.45576096 9,3.00247329 L9,7 L12.9975267,7 C13.5116309,7 13.9353477,7.38272018 13.9932556,7.88266744 L14,8 C14,8.55228475 13.544239,9 12.9975267,9 L8.999,9 L9,12.9975267 C9,13.5116309 8.61727982,13.9353477 8.11733256,13.9932556 L8,14 C7.44771525,14 7,13.544239 7,12.9975267 L6.999,9 L3.00247329,9 C2.44882258,9 2,8.55613518 2,8 C2,7.44771525 2.45576096,7 3.00247329,7 L7,7 L7,3.00247329 C7,2.44882258 7.44386482,2 8,2 Z\\"></path>    </g></svg>"
                            >
                              <Icon
                                className="emotion-8 emotion-9"
                                icon="<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <path d=\\"M8,2 C8.55228475,2 9,2.45576096 9,3.00247329 L9,7 L12.9975267,7 C13.5116309,7 13.9353477,7.38272018 13.9932556,7.88266744 L14,8 C14,8.55228475 13.544239,9 12.9975267,9 L8.999,9 L9,12.9975267 C9,13.5116309 8.61727982,13.9353477 8.11733256,13.9932556 L8,14 C7.44771525,14 7,13.544239 7,12.9975267 L6.999,9 L3.00247329,9 C2.44882258,9 2,8.55613518 2,8 C2,7.44771525 2.45576096,7 3.00247329,7 L7,7 L7,3.00247329 C7,2.44882258 7.44386482,2 8,2 Z\\"></path>    </g></svg>"
                              >
                                <WrapperStyled
                                  className="emotion-8 emotion-9"
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <path d=\\"M8,2 C8.55228475,2 9,2.45576096 9,3.00247329 L9,7 L12.9975267,7 C13.5116309,7 13.9353477,7.38272018 13.9932556,7.88266744 L14,8 C14,8.55228475 13.544239,9 12.9975267,9 L8.999,9 L9,12.9975267 C9,13.5116309 8.61727982,13.9353477 8.11733256,13.9932556 L8,14 C7.44771525,14 7,13.544239 7,12.9975267 L6.999,9 L3.00247329,9 C2.44882258,9 2,8.55613518 2,8 C2,7.44771525 2.45576096,7 3.00247329,7 L7,7 L7,3.00247329 C7,2.44882258 7.44386482,2 8,2 Z\\"></path>    </g></svg>",
                                    }
                                  }
                                  onClick={[Function]}
                                >
                                  <span
                                    className="emotion-9 emotion-13 emotion-14"
                                    dangerouslySetInnerHTML={
                                      Object {
                                        "__html": "<svg width=\\"16px\\" height=\\"16px\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlnsXlink=\\"http://www.w3.org/1999/xlink\\">    <g stroke=\\"none\\" stroke-width=\\"1\\">        <path d=\\"M8,2 C8.55228475,2 9,2.45576096 9,3.00247329 L9,7 L12.9975267,7 C13.5116309,7 13.9353477,7.38272018 13.9932556,7.88266744 L14,8 C14,8.55228475 13.544239,9 12.9975267,9 L8.999,9 L9,12.9975267 C9,13.5116309 8.61727982,13.9353477 8.11733256,13.9932556 L8,14 C7.44771525,14 7,13.544239 7,12.9975267 L6.999,9 L3.00247329,9 C2.44882258,9 2,8.55613518 2,8 C2,7.44771525 2.45576096,7 3.00247329,7 L7,7 L7,3.00247329 C7,2.44882258 7.44386482,2 8,2 Z\\"></path>    </g></svg>",
                                      }
                                    }
                                    onClick={[Function]}
                                  />
                                </WrapperStyled>
                              </Icon>
                            </IconStyled>
                          </button>
                        </CounterButtonStyled>
                      </CounterControl>
                    </span>
                  </CounterStyled>
                </span>
              </WrapperStyled>
            </Component>
          </OmittedComponent>
        </Component>
      </IMask(Component)>
    </TextFieldMasked>
  </Component>
</OmittedComponent>
`;
